File: lua/bob/bob.lua
local bob = {}

function bob.hey(text)
  if #text == 0 then return 'Fine, be that way.' end

  if text:sub(-1) == '?' then return 'Sure' end

  if text:upper() == text then return 'Whoa, chill out!' end

  return 'Whatever'
end

return bob

File: ruby/bob/bob.rb
# Bob simulates a lackadaisical teenager.
module Bob
  def self.hey(remark)
    remark.strip!

    if remark.empty?
      'Fine. Be that way!'
    elsif remark =~ /[a-z]/i && remark == remark.upcase
      'Whoa, chill out!'
    elsif remark.end_with?('?')
      'Sure.'
    else
      'Whatever.'
    end
  end
end

module BookKeeping
  VERSION = 1
end
